src/ring_theory/ideal/operations.lean

```diff
@@ -1316,7 +1316,30 @@ lemma quotient_map_comp_mk {J : ideal R} {I : ideal S} {f : R →+* S} (H : J 
   (quotient_map I f H).comp (quotient.mk J) = (quotient.mk I).comp f :=
 ring_hom.ext (λ x, by simp only [function.comp_app, ring_hom.coe_comp, ideal.quotient_map_mk])
 
-/-- `H` and `h` are kept as separate hypothesis since H is used in constructing the quotient map -/
+/-- If `f : R ≃+* S` is a ring isomorphism and `I : ideal R`, then `map f (map f.symm) = I`. -/
+lemma map_of_equiv (I : ideal R) (f : R ≃+* S) : (I.map f.to_ring_hom).map f.symm.to_ring_hom = I :=
+by simp [map_map, map_id]
+
+/-- If `f : R ≃+* S` is a ring isomorphism and `I : ideal R`, then `comap f.symm (comap f) = I`. -/
+lemma comap_of_equiv (I : ideal R) (f : R ≃+* S) :
+  (I.comap f.symm.to_ring_hom).comap f.to_ring_hom = I :=
+by simp [comap_comap, comap_id]
```

Comment:
`map_id` and `comap_id` are already simp lemmas:

```suggestion
/-- If `f : R ≃+* S` is a ring isomorphism and `I : ideal R`, then `map f (map f.symm) = I`. -/
lemma map_of_equiv (I : ideal R) (f : R ≃+* S) : (I.map f.to_ring_hom).map f.symm.to_ring_hom = I :=
by simp [map_map]

/-- If `f : R ≃+* S` is a ring isomorphism and `I : ideal R`, then `comap f.symm (comap f) = I`. -/
lemma comap_of_equiv (I : ideal R) (f : R ≃+* S) :
  (I.comap f.symm.to_ring_hom).comap f.to_ring_hom = I :=
by simp [comap_comap]
```

What are your thoughts on making `map_map` and `comap_comap` simp lemmas too? `list.map_map` is a simp lemma, so I think it would be reasonable. Then you wouldn't need this lemma at all, as `simp` would solve it.

Reply:
It seems to me a good idea, but I don't know the good habits for `simp` lemmas :)